src/bridge/cockpitpcpmetrics.c

```diff
@@ -0,0 +1,1017 @@
+          goto out;
+        }
+
+      if (pmLookupDesc (info->id, &info->desc) < 0)
+        {
+          g_warning ("%s: no such metric: %s (%s)", self->name, info->name, pmErrStr (self->context));
+          goto out;
+        }
+
+      if (units)
+        {
+          if (type == NULL)
+            type = "number";
+
+          if (cockpit_pmParseUnitsStr (units, &info->units_buf, &info->factor) < 0)
+            {
+              g_warning ("%s: failed to parse units: %s", self->name, units);
+              goto out;
+            }
+
+          if (!units_convertible (&info->desc.units, &info->units_buf))
+            {
+              g_warning ("%s: can't convert metric %s to units %s", self->name, info->name, units);
+              goto out;
+            }
+
+          if (info->factor != 1.0 || !units_equal (&info->desc.units, &info->units_buf))
+            info->units = &info->units_buf;
+        }
+
+      if (!info->units)
+        {
+          info->units = &info->desc.units;
+          info->factor = 1.0;
+        }
+
+      if (g_strcmp0 (type, "number") == 0)
+        {
+          int dt = info->desc.type;
+          if (!(dt == PM_TYPE_32 || dt == PM_TYPE_U32 ||
+                dt == PM_TYPE_64 || dt == PM_TYPE_U64 ||
+                dt == PM_TYPE_FLOAT || dt == PM_TYPE_DOUBLE))
+            {
+              g_warning ("%s: metric %s is not a number", self->name, info->name);
+              goto out;
+            }
+        }
+      else if (g_strcmp0 (type, "string") == 0)
+        {
+          if (info->desc.type != PM_TYPE_STRING)
+            {
+              g_warning ("%s: metric %s is not a string", self->name, info->name);
+              goto out;
+            }
+        }
+      else if (type != NULL)
+        {
+          g_warning ("%s: unsupported type %s", self->name, type);
+          goto out;
+        }
+
+      if (g_strcmp0 (semantics, "counter") == 0)
+        {
+          if (info->desc.sem != PM_SEM_COUNTER)
+            {
+              g_warning ("%s: metric %s is not a counter", self->name, info->name);
+              goto out;
+            }
+        }
+      else if (g_strcmp0 (semantics, "instant") == 0)
+        {
+          if (info->desc.sem != PM_SEM_INSTANT)
+            {
+              g_warning ("%s: metric %s is not instantaneous", self->name, info->name);
+              goto out;
+            }
+        }
+      else if (g_strcmp0 (semantics, "discrete") == 0)
+        {
+          if (info->desc.sem != PM_SEM_DISCRETE)
+            {
+              g_warning ("%s: metric %s is not discrete", self->name, info->name);
+              goto out;
+            }
+        }
+      else if (semantics != NULL)
+        {
+          g_warning ("%s: unsupported semantics %s", self->name, semantics);
+          goto out;
+        }
+
+      self->pmidlist[i] = info->id;
+
+      if (info->desc.indom != PM_INDOM_NULL)
```

Comment:
What happens when instances or omit-instances are specified when the metric has no instances? What should happen?